src/Features/Core/Portable/ExtractMethod/MethodExtractor.Analyzer.cs

```diff
@@ -122,9 +122,6 @@ public async Task<AnalyzerResult> AnalyzeAsync()
                 var returnTypeHasAnonymousType = returnTypeTuple.hasAnonymousType;
                 var awaitTaskReturn = returnTypeTuple.awaitTaskReturn;
 
-                // create new document
-                var newDocument = await CreateDocumentWithAnnotationsAsync(_semanticDocument, parameters, CancellationToken).ConfigureAwait(false);
```

Comment:
AnalyzerResult now no longer changes as the work is happening.  it is simply computed up front and all the data in it stays fixed (making things much clearer/easier).